bazarr/main.py

```diff
@@ -1917,6 +1918,49 @@ def manual_get_subtitle():
         pass
 
 
+@route(base_url + 'manual_upload_subtitle', method='POST')
+@custom_auth_basic(check_credentials)
+def perform_manual_upload_subtitle():
+    authorize()
+    ref = request.environ['HTTP_REFERER']
+
+    episodePath = request.forms.get('episodePath')
+    sceneName = request.forms.get('sceneName')
+    language = request.forms.get('language')
+    upload = request.files.get('upload')
+    sonarrSeriesId = request.forms.get('sonarrSeriesId')
+    sonarrEpisodeId = request.forms.get('sonarrEpisodeId')
+    title = request.forms.get('title')
+
+    _, ext = os.path.splitext(upload.filename)
+
+    if ext not in SUBTITLE_EXTENSIONS:
+        raise ValueError('A subtitle of an invalid format was uploaded.')
+    
+    try:
+        result = manual_upload_subtitle(path=episodePath, 
+                                        language=language, 
+                                        title=title, 
+                                        scene_name=sceneName, 
+                                        media_type='series',
+                                        subtitle=upload)
+        
+        if result is not None:
+            message = result[0]
+            path = result[1]
+            language_code = language
+            provider = "manual"
+            score = 360
+            history_log(2, sonarrSeriesId, sonarrEpisodeId, message, path, language_code, provider, score)
```

Comment:
I would define a new action here (0=delete, 1=download, 2=manual_download, 3=upgrade). $ would be manual upload. As we only upgrade those with action 1, 2 or 3, manually uploaded subs will never be upgraded. You can also have a dedicated icon in history view.